lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -455,7 +455,16 @@ sub prepare_job_results {
     my ($self, $jobs) = @_;
     my %archs;
     my %results;
-    my $aggregated = {none => 0, passed => 0, failed => 0, incomplete => 0, scheduled => 0, running => 0, unknown => 0};
+    my $aggregated = {
+        none         => 0,
+        passed       => 0,
+        failed       => 0,
+        not_complete => 0,
+        aborted      => 0,
+        scheduled    => 0,
+        running      => 0,
+        unknown      => 0
+    };
```

Comment:
feels like an arbitrary list of "statuses" we have here. Do we even need to initialize the hash at all? If it is for conducting `$aggregated->{foo}++`, this also works in perl on uninitialized hashes.

Reply:
Yes, because the value must be `0` and not `undef` when there are no jobs of a certain type (at least for passed and failed). We could move the code for catching that case into the template but it is likely better inside the controller.

Reply:
where do we rely on values being zero instead of undef?

Reply:
In the template consuming this hash, at least for passed and failed.